app/js/enhancedDate.js

```diff
@@ -1,5 +1,133 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function (date) {
+// still maintains access to all globals
+  
+  var setDate = {};
+
+//Stores the passed-in date.  
+  setDate = function (date) {
+      var myDate = date;
+    
+//Do some type-checking to make sure you have a valid type.    
+      if(typeof myDate !== null)
```

Comment:
The typeof operation will never return null, it will only ever return a string value. If you want to check if something is a Date object, use if (myDate instanceof Date) instead. The instanceof operator is what you want to use when checking if something is a specific kind of object and the thing you check against is the constructor for that kind of object.